### ISSUE TITLE
False positive in closure_end_indentation

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#3704]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
/Users/marcelofabri/SwiftLint/file.swift:10:1: warning: Closure End Indentation Violation: Closure end should have the same indentation as the line that started it. Expected 16, got 0. (closure_end_indentation)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.43.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - closure_end_indentation
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

No

* Which Xcode version are you using (check `xcodebuild -version`)?

12.5.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
Observable.combineLatest(
    mapSectionProviders(context: context)
        .map { provider -> Observable<HomeSection> in
            provider.fetch(context: context)
                .startWith(.empty)
                .catchAndReturn(.empty)
        }
).map { sections in
    sections.filter { !$0.items.isEmpty }
}
```

**Comment by iSevenDays** [comment]
I have the same issue:

Example
```
file.swift
Section(header: Text("dinner")
        .font(.system(size: 17, weight: .semibold)),
        footer: SomeView(
          error: "",
          isTimeValid: true)) {
            DatePicker("Start", selection: Date() displayedComponents: .hourAndMinute)
              DatePicker("End" selection: Date() displayedComponents: .hourAndMinute)
} <-- no error if it is here
          } <-- error if it is here, but it should be here
```

The swiftlint moves the latest brace to the start of the Section line, not to the start of isTimeValid line

**Comment by RobertHaworth** [comment]
I am also seeing this issue currently.

`APIManager
            .Inventory
            .QuickSearch(toolId: toolId)
            .execute { response in
                switch response {
                case let .success(value):
                    /* do work */
                case let .failure(error):
                    /* report error */
                }
        }      <=== Error here, correct location as per ctrl+i
} <=== no error here.
`

**Comment by SimplyDanny** [comment]
All these issue have been resolved by #6109. Please open new tickets if that's not the case.
